runtime/lua/vim/version.lua

```diff
@@ -168,6 +170,7 @@ function M._version(version, strict) -- Adapted from https://github.com/folke/la
   end
 
   if not strict then -- TODO: add more "scrubbing".
+    --- @cast version string
```

Comment:
Hmm, I'm surprised LuaLS isn't properly type narrowing here.

I think this is the longstanding bug where if the RHS contains the LHS, then LuaLS gets confused what the type is.

Reply:
Yeah the issue is that `Version` is a technically a table, but LuaLS doesn't realize that and so it thinks that here `version` is of type `Version|string` and so it complains about the `match`.